### PR TITLE
Declutter multipart functionality in api.

### PR DESCRIPTION
**Pull request by BrandtHill**
Remove buggy and now unused code that was causing some issues when moving from .5 rc to latest
Make request_multipart useful, removing duplicate code to make multipart and file handling easier

Additionally:

- fix spec for request_multipart
- fix bug where execute_webhook and edit_webhook message wasn’t casting to `Message` struct
- make internal function private
- silently handle combining `:embed` and `:embeds` similarly to `:file` and `:files`
- make file form name `files[n]` where `n` begins at 0 according to most recent docs
- support files for all applicable message editing functions

### CODE REVIEW

File: lib/nostrum/api.ex [hunk 1]
```diff
@@ -4208,12 +4075,19 @@ defmodule Nostrum.Api do
   @spec request_multipart(atom(), String.t(), any, keyword() | map()) :: {:ok} | error
   def request_multipart(method, route, body, params \\ []) do
     boundary = generate_boundary()
+    body = combine_files(body)
+    files = body.files
+
+    json =
+      body
+      |> Map.delete(:files)
+      |> Jason.encode_to_iodata!()
```
Comment (Th3-M4jor) [inline lib/nostrum/api.ex:4084]: Could probably simplify this to
```elixir
{files, body} = combine_files(body) |> Map.pop!(:files)

json = body |> Jason.encode_to_iodata!()
```

Other than that, this looks good to me.

**Review by Th3-M4jor — CHANGES_REQUESTED** [review]

**Comment by BrandtHill** [comment]
Btw please test these. I’ve been writing this on my phone.

**Comment by BrandtHill** [comment]
Fixed the create_interaction_response, could you test again?

**Comment by Th3-M4jor** [comment]
Tested and it works, but now `Api.create_followup_message!/3` is throwing a dialyzer error on my machine. Saying the function call will not succeed.

**Comment by BrandtHill** [comment]
> Tested and it works, but now `Api.create_followup_message!/3` is throwing a dialyzer error on my machine. Saying the function call will not succeed.

I think it was the request_multipart spec which I just fixed

**Comment by Th3-M4jor** [comment]
I found the issue, `Api.execute_webhook` isn't decoding the response into a Message struct, it needs a `|> handle_request_with_decode({:struct, Message})`

**Review by jchristgit — APPROVED** [review]
Deleted code ✅ 

Thanks!

**Comment by jchristgit** [comment]
Thanks!

**Pull request merged**